### ISSUE TITLE
Cannot access IMAP spoolfile folders from other IMAP accounts

### ISSUE THREAD

**Issue by chillycuz** [neomutt/neomutt#553]
Please note: I originally submitted this to the core mutt maintainers (https://dev.mutt.org/trac/ticket/3933) because I thought it might be something to do with the new folder management. After exploring the issue and finally trying mutt, where actual behavior matched expected behavior, it seems as though the issue lies in neomutt. There is more information in that work tag.

OS: Arch Linux up to do; Neomutt 1.8.2

Basically I have two IMAP accounts. One is set as the spoolfile in muttrc, which I will call the "spoolfile account". I will call the other "Account2".

The **expected behavior** is that after pressing "c" from Account2 to change folder and then selecting a folder in Spoolfile Account, the mailbox will change to the new folder. Similarly, after pressing "s" to save an email to a folder in Spoolfile Account and selecting that folder, the email will be saved in the Spoolfile Account folder.

Since 1.8.2 update from Arch Linux AUR, the **actual behavior** is that neither of these things happen and the Spoolfile Account folder is not recognized.

Case 1: Changing folders From the spoolfile account (imaps://cuz@blah...), I type "c", which takes me to the file browser where it lists the IMAP folders. In the listings under the main mailbox (imaps://cuz@blah....), the folders are listed as ("=folder1", "=folder2", etc.). If I select one of these and press enter, everything works as expected, ie, the folder opens. However, if I do exactly the same thing from Account2, then I get an error message telling me that "Mailbox does not exist."

Case 2: Saving emails From the spoolfile account, I type "s" to move a message, which takes me to the file browser where it lists the local folders. I then press "TAB" to view the IMAP folders. If I then select one of the folders in the spoolfile account and press enter, everything works as expected, ie, the email is saved. However, if I do exactly the same thing from Account2, then I get a message asking me if I would like to "Create mailbox".

Case 3: Using key bindings to save/move emails As explained in the original message, in either account (spoolfile account or Account2), if I use a keybinding that specifies a particular folder, e.g. ",a" with the binding above, the email is saved to the spoolfile account folder as expected.

Thanks for taking a look. I love neomutt and would like to switch back from vanilla mutt. :)

**Comment by flatcap** [comment]
The IMAP code is flaky, but it may be a while before we risk making significant changes.
I'm going to close this as "unfixable" for now.

**Comment by flatcap** [comment]
I have four imap accounts and I can switch (and copy) between them as expected.

Please can you try the latest version of NeoMutt and see if your problem still occurs.

**Comment by chillycuz** [comment]
Thanks for taking a look. (I've been moving and having been paying attention to this issue.) I just installed neomutt from AUR and continue to have the same problem, which does not exist in core mutt. I just tried copying from one imap account inbox to another. That works. I also have no problem copying it to a local Maildir folder. The problem arises when I try to copy or move an email to a folder on the remote imap server, e.g., "=oldemail".
